common/src/main/java/org/wildfly/swarm/ts/common/DockerRunner.java

```diff
@@ -0,0 +1,167 @@
+        this.logWait = logWait;
+    }
+
+    public String getContainerImage() {
+        return containerImage;
+    }
+
+    public void setContainerImage(String containerImage) {
+        this.containerImage = containerImage;
+    }
+
+    public String getLogWait() {
+        return logWait;
+    }
+
+    public void setLogWait(String logWait) {
+        this.logWait = logWait;
+    }
+
+    public void addEnvVar(String key, String value) {
+        environmentVariables.put(key, value);
+    }
+
+    public void addPort(String port) {
+        this.ports.add(port);
+    }
+
+    public List<String> getPorts() {
+        return this.ports;
+    }
+
+    public long getTimeoutMillis() {
+        return timeoutMillis;
+    }
+
+    public void setTimeoutMillis(long timeoutMillis) {
+        this.timeoutMillis = timeoutMillis;
+    }
+
+    public void start() throws Exception {
+
+        containerId = UUID.randomUUID().toString();
+        ProcessBuilder dockerStartProcessBuilder = new ProcessBuilder();
+
+        List<String> parameters = new ArrayList<>();
+
+        parameters.add("docker");
+        parameters.add("run");
+
+        for (String port : ports) {
+            parameters.add("-p");
+            parameters.add(port);
+        }
+
+        for (Entry<String, String> envVar : environmentVariables.entrySet()) {
+            parameters.add("-e");
+            parameters.add("\"" + envVar.getKey() + "=" + envVar.getValue() + "\"");
+        }
+
+        parameters.add("--name");
+        parameters.add(containerId);
+        parameters.add(containerImage);
+
+        dockerStartProcessBuilder.redirectErrorStream(true);
+        dockerStartProcessBuilder.command(parameters);
+
+        Process process = dockerStartProcessBuilder.start();
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+
+        executor.execute(() -> {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line = null;
+                while ((line = reader.readLine()) != null) {
+                    LOGGER.info(line);
+
+                    if (logWait != null && line.contains(logWait)) {
+                        break;
+                    }
+                }
+                return;
+            } catch (IOException e) {
+                return;
+            }
+        });
+
+        executor.shutdown();
+
+        if (!executor.awaitTermination(timeoutMillis, TimeUnit.MILLISECONDS)) {
+            process.destroy();
+        }
+    }
+
+    public void stop() throws IOException, InterruptedException {
+        ProcessBuilder stopProcessBuilder = new ProcessBuilder();
+        stopProcessBuilder.command("docker", "stop", containerId);
+        stopProcessBuilder.start().waitFor(10, TimeUnit.SECONDS);
+
+        ProcessBuilder rmProcessBuilder = new ProcessBuilder();
+        rmProcessBuilder.command("docker", "rm", containerId);
```

Comment:
Should probably start the container with `--rm`, so that it's deleted automatically when stopped.